File: programme/perimeter-of-square/perimeter-of-square.c
#include <stdio.h>
#include <stdlib.h> /* for atof function */

int main()
{
    double side, perimeter;
    char sidestr[32];

    printf("Enter length of side: ");
    for (;;) { /* get user input until input is valid */
        scanf("%s", sidestr);
        side = atof(sidestr);
        if (side <= 0) {
            printf("Invalid input.\nEnter length of side: ");
            continue;
        }
        else
            break;
    }
    perimeter = 4 * side;
    printf("Perimeter: %g\n", perimeter);
    return 0;
}

File: programme/perimeter-of-square/perimeter-of-square.cs
// Write a programme to find perimeter of square
using System;

public class PerimeterOfSquare
{
    public static void Main(string[] args)
    {
        Console.Write("Side : ");
        string input = Console.ReadLine();
        int side = 0;

        // verifies valid input and passes converted int to side variable
        if (int.TryParse(input, out side))
        {
            Console.WriteLine("Perimeter : " + (side * 4));
        }
        else
        {
            Console.WriteLine("\n*** Invalid input !!! ***");
        }
    }
}
